.github/workflows/go.yml

```diff
@@ -9,16 +9,22 @@ on:
   pull_request:
     branches: [ "main" ]
 
+permissions:
+  contents: read
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
 
     - name: Set up Go
       uses: actions/setup-go@v4
       with:
         go-version-file: 'go.mod'
+        cache: false
```

Comment:
Bit annoying that we have to disable this, but given this is a very small codebase with almost no deps it should be fine.